src/user/user.controller.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpException,
+  HttpStatus,
+  Param,
+  Post,
+  Query,
+  Request as Req,
+  UseGuards,
+} from '@nestjs/common';
+import { User } from './user.model';
+import { UserService } from './user.service';
+import { CURRENT_USER } from 'src/constants';
+import { Request } from 'express';
+import { Public } from 'src/decorators/public.decorator';
+import { AuthService } from 'src/auth/auth.service';
+import { LocalAuthGuard } from 'src/guards/local-auth.guard';
+import { UserDto } from 'src/dtos/user.dto';
+
+@Controller()
+export class UserController {
+  constructor(
+    private userService: UserService,
+    private readonly authService: AuthService,
+  ) {}
+
+  @Public()
+  @UseGuards(LocalAuthGuard)
+  @Post('login')
+  login(@Req() req: Request) {
+    console.log('req.user', req.user);
+    return this.authService.login(req.user);
+  }
+
+  /**
+   * Create and update route
+   */

```

Comment:
Guard all endpoint as well.

Reply:
I am already using Global JwtAuthGuard that protects all the routes. [https://github.com/AimenBaseer/nestjs-app/blob/cf1aa6513759785ffc11f638d4cce786fb5b13cf/src/app.module.ts](url)